chat/message/theme.go

```diff
@@ -122,43 +125,49 @@ type Theme struct {
 	names     *Palette
 }
 
-func (t Theme) ID() string {
-	return t.id
+func (theme Theme) ID() string {
+	return theme.id
 }
 
 // Colorize name string given some index
-func (t Theme) ColorName(u *User) string {
-	if t.names == nil {
+func (theme Theme) ColorName(u *User) string {
+	if theme.names == nil {
 		return u.Name()
 	}
 
-	return t.names.Get(u.colorIdx).Format(u.Name())
+	return theme.names.Get(u.colorIdx).Format(u.Name())
 }
 
 // Colorize the PM string
-func (t Theme) ColorPM(s string) string {
-	if t.pm == nil {
+func (theme Theme) ColorPM(s string) string {
+	if theme.pm == nil {
 		return s
 	}
 
-	return t.pm.Format(s)
+	return theme.pm.Format(s)
 }
 
 // Colorize the Sys message
-func (t Theme) ColorSys(s string) string {
-	if t.sys == nil {
+func (theme Theme) ColorSys(s string) string {
+	if theme.sys == nil {
 		return s
 	}
 
-	return t.sys.Format(s)
+	return theme.sys.Format(s)
 }
 
 // Highlight a matched string, usually name
-func (t Theme) Highlight(s string) string {
-	if t.highlight == nil {
+func (theme Theme) Highlight(s string) string {
+	if theme.highlight == nil {
 		return s
 	}
-	return t.highlight.Format(s)
+	return theme.highlight.Format(s)
+}
+
+// Timestamp formats and colorizes the timestamp.
+func (theme Theme) Timestamp(t time.Time) string {
+	// TODO: Change this per-theme? Or config?
+	return theme.sys.Format(t.Format("2006-01-02 15:04:05 UTC"))
```

Comment:
I don't particularly like the nearly-static yyyy-mm-dd & truly-static " UTC" strings on every line, but I find the reasoning for their inclusion to be adequate.

Although, I wonder if the reasoning would still hold after the user could explicitly set a timezone, or it were guessed by the incoming IP address... as the user would 'know' the time zone, and not need reminding every line.

I don't know how expensive the ```time.Format``` function is, but calling it for every timestamp trades runtime performance for code clarity.

Reply:
GeoIP and things like that is a lot more work, so really it's a question whether you think this should be merged now or if it should be blocked until someone someday writes the code to automatically detect timezones. :)

`time.Format` is not expensive enough to worry, ~everything else we do is much more expensive.

Reply:
You can use the `TZ` environment variable, if that being passed from a SSH client, like:
`ssh -o SendEnv=TZ ...`